app/src/test/java/org/mozilla/fenix/settings/address/ext/AddressKtTest.kt

```diff
@@ -0,0 +1,145 @@
+        val addr = generateAddress()
+
+        val result = addr.toItemText()
+
+        assertEquals("${addr.givenName} ${addr.additionalName} ${addr.familyName}", result.label)
+        assertEquals(addr.streetAddress, result.description)
+    }
+
+    @Test
+    fun `WHEN middle name is missing THEN given and family are combined`() {
+        val addr = generateAddress(additional = false)
+
+        val result = addr.toItemText()
+
+        assertEquals("${addr.givenName} ${addr.familyName}", result.label)
+        assertEquals(addr.streetAddress, result.description)
+    }
+
+    @Test
+    fun `WHEN only family and middle name are available THEN family is used`() {
+        val addr = generateAddress(given = false)
+
+        val result = addr.toItemText()
+
+        assertEquals(addr.familyName, result.label)
+        assertEquals(addr.streetAddress, result.description)
+    }
+
+    @Test
+    fun `WHEN name is missing THEN street address and address level 3 are used`() {
+        val addr = generateAddress(given = false, additional = false, family = false)
+
+        val result = addr.toItemText()
+
+        assertEquals(addr.streetAddress, result.label)
+        assertEquals(addr.addressLevel3, result.description)
+    }
+
+    @Test
+    fun `WHEN everything but tel and email are missing THEN tel and email are used`() {
+        val addr = generateAddress(
+            given = false,
+            additional = false,
+            family = false,
+            org = false,
+            streetAddr = false,
+            addr3 = false,
+            addr2 = false,
+            addr1 = false,
+            postalCode = false,
+            country = false
+        )
+
+        val result = addr.toItemText()
+
+        assertEquals(addr.tel, result.label)
+        assertEquals(addr.email, result.description)
+    }
+
+    @Test
+    fun `WHEN everything but email is missing THEN email is label and description is empty`() {
+        val addr = generateAddress(
+            given = false,
+            additional = false,
+            family = false,
+            org = false,
+            streetAddr = false,
+            addr3 = false,
+            addr2 = false,
+            addr1 = false,
+            postalCode = false,
+            country = false,
+            tel = false
+        )
+
+        val result = addr.toItemText()
+
+        assertEquals(addr.email, result.label)
+        assertEquals("", result.description)
+    }
+
+    @Test
+    fun `GIVEN multiline street address THEN joined as single line`() {
+        val addr = generateAddress().copy(
+            streetAddress = """
+                line1
+                line2
+                line3
+            """.trimIndent()
+        )
+
+        val result = addr.toItemText()
+
+        assertEquals("line1 line2 line3", result.description)
+    }
+
+    private fun generateAddress(
+        given: Boolean = true,
```

Comment:
Couldn't we have just specify the default string values for all of these parameters instead of using booleans?

Reply:
Yeah, that's definitely better. Thanks for the suggestion

Reply:
Gonna request that we just allow that this function takes in string values for each of the parameter. This type of function can generally be reused in many other tests so it would be best if it was robust.